tests/test_openclhpp.cpp

```diff
@@ -4115,4 +4119,105 @@ void testEnqueueAcquireExternalMemObjects(void) {}
+        case CL_IMAGE_REQUIREMENTS_SLICE_PITCH_ALIGNMENT_EXT:
+#endif
+        case CL_IMAGE_REQUIREMENTS_SIZE_EXT:
+        {
+            size_t ret = 4;
+            if (param_value_size == sizeof(ret) && param_value) {
+                *static_cast<size_t *>(param_value) = ret;
+            }
+
+            if (param_value_size_ret) {
+                *param_value_size_ret = sizeof(ret);
+            }
+
+            return CL_SUCCESS;
+        }
+        case CL_IMAGE_REQUIREMENTS_MAX_WIDTH_EXT:
+        case CL_IMAGE_REQUIREMENTS_MAX_HEIGHT_EXT:
+        case CL_IMAGE_REQUIREMENTS_MAX_DEPTH_EXT:
+        case CL_IMAGE_REQUIREMENTS_MAX_ARRAY_SIZE_EXT:
+        {
+            cl_uint ret = TEST_UINT_VALUE;
+            if (param_value_size == sizeof(ret) && param_value) {
+                *static_cast<cl_uint *>(param_value) = ret;
+            }
+
+            if (param_value_size_ret) {
+                *param_value_size_ret = sizeof(ret);
+            }
+
+            return CL_SUCCESS;
+        }
+        default: break;
+    }
+    TEST_FAIL();
+    return CL_INVALID_OPERATION;
+}
+
+void testTemplateGetImageRequirementsInfo()
+{
+    cl::Context context(make_context(0));
+    cl_device_id device_expect = make_device_id(0);
+    int device_refcount = 1;
+
+    prepare_deviceRefcounts(1, &device_expect, &device_refcount);
+
+    clGetContextInfo_StubWithCallback(clGetContextInfo_device);
+    clGetDeviceInfo_StubWithCallback(clGetDeviceInfo_platform);
+    clGetPlatformInfo_StubWithCallback(clGetPlatformInfo_version_2_0);
+    clReleaseContext_ExpectAndReturn(make_context(0), CL_SUCCESS);
+
+    clGetImageRequirementsInfoEXT_StubWithCallback(clGetImageRequirementsInfoEXT_GetInfo);
+
+    auto info0 = context.getImageRequirementsInfoExt<cl::ImageRequirementsInfoExt::RowPitchAlign>();
+    TEST_ASSERT_EQUAL(info0, TEST_SIZE_TYPE_VALUE);
+
+    auto info1 = context.getImageRequirementsInfoExt<cl::ImageRequirementsInfoExt::BaseAddAlign>();
+    TEST_ASSERT_EQUAL(info1, TEST_SIZE_TYPE_VALUE);
+
+    auto info2 = context.getImageRequirementsInfoExt<cl::ImageRequirementsInfoExt::Size>();
+    TEST_ASSERT_EQUAL(info2, TEST_SIZE_TYPE_VALUE);
+
+#if defined(cl_ext_image_from_buffer)
+    auto info3 = context.getImageRequirementsInfoExt<cl::ImageRequirementsInfoExt::SlicePitchAlign>();
+    TEST_ASSERT_EQUAL(info3, TEST_SIZE_TYPE_VALUE);
+#endif
+
+    auto info4 = context.getImageRequirementsInfoExt<cl::ImageRequirementsInfoExt::NaxWidth>();
+    TEST_ASSERT_EQUAL(info4, TEST_UINT_VALUE);
+
+    auto info5 = context.getImageRequirementsInfoExt<cl::ImageRequirementsInfoExt::MaxHeight>();
+    TEST_ASSERT_EQUAL(info5, TEST_UINT_VALUE);
+
+    auto info6 = context.getImageRequirementsInfoExt<cl::ImageRequirementsInfoExt::MaxDepth>();
+    TEST_ASSERT_EQUAL(info6, TEST_UINT_VALUE);
+
+    auto info7 = context.getImageRequirementsInfoExt<cl::ImageRequirementsInfoExt::MaxArraySize>();
+    TEST_ASSERT_EQUAL(info7, TEST_UINT_VALUE);
+}
+#endif
```

Comment:
This needs an `#else` so the symbol is still defined even if the extension is not in the headers.

```suggestion
#else
void testTemplateGetImageRequirementsInfo() {}
#endif
```

Reply:
done :)